### PR TITLE
Let `--bisect=~<old>..<new>` find the first fixing commit

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Comment by fmeum** [comment]
@meteorcloudy

File: README.md [hunk 1]
```diff
@@ -130,17 +130,21 @@ This will show you which flags can safely enabled, and which flags require a mig
 ### --bisect
 
 `--bisect` flag allows you to bisect Bazel versions to find which version introduced a build failure. You can specify the range of versions to bisect with `--bisect=<GOOD>..<BAD>`, where GOOD is the last known working Bazel version and BAD is the first known non-working Bazel version. Bazelisk uses [GitHub's compare API](https://docs.github.com/en/rest/commits/commits#compare-two-commits) to get the list of commits to bisect. When GOOD is not an ancestor of BAD, GOOD is reset to their merge base commit.
+The meaning of GOOD and BAD can be reversed by prefixing the range with `~`, e.g. `--bisect=!6.0.0..HEAD` will find the first version 6.0.0 and HEAD that *fixes* the build.
```
Comment (meteorcloudy) [inline README.md:133]: `--bisect=!6.0.0..HEAD` -> `--bisect=~6.0.0..HEAD`?

File: core/core.go [hunk 2]
```diff
@@ -165,14 +165,14 @@ func RunBazeliskWithArgsFuncAndConfigAndOut(argsFunc ArgsFunc, repos *Repositori
 	} else if len(args) > 0 && strings.HasPrefix(args[0], "--bisect") {
 		// When --bisect is present, we run the bisect logic.
 		if !strings.HasPrefix(args[0], "--bisect=") {
-			return -1, fmt.Errorf("Error: --bisect must have a value. Expected format: '--bisect=<good bazel commit>..<bad bazel commit>'")
+			return -1, fmt.Errorf("Error: --bisect must have a value. Expected format: '--bisect=[!]<good bazel commit>..<bad bazel commit>'")
```
Comment (meteorcloudy) [inline core/core.go:168]: ~

File: core/core.go [hunk 3]
```diff
@@ -165,14 +165,14 @@ func RunBazeliskWithArgsFuncAndConfigAndOut(argsFunc ArgsFunc, repos *Repositori
...
 		}
 		value := args[0][len("--bisect="):]
 		commits := strings.Split(value, "..")
 		if len(commits) == 2 {
 			bisect(commits[0], commits[1], args[1:], bazeliskHome, repos, config)
 		} else {
-			return -1, fmt.Errorf("Error: Invalid format for --bisect. Expected format: '--bisect=<good bazel commit>..<bad bazel commit>'")
+			return -1, fmt.Errorf("Error: Invalid format for --bisect. Expected format: '--bisect=[!]<good bazel commit>..<bad bazel commit>'")
```
Comment (meteorcloudy) [inline core/core.go:175]: ~

File: core/core.go [hunk 4]
```diff
@@ -766,47 +766,47 @@ func sendRequest(url string, config config.Config) (*http.Response, error) {
...
+
 		if len(compResp.Commits) == 0 {
 			break
 		}
 
 		mergeBaseCommit := compResp.MergeBaseCommit.SHA
 		if mergeBaseCommit != compResp.BaseCommit.SHA {
-			fmt.Printf("The good Bazel commit is not an ancestor of the bad Bazel commit, overriding the good Bazel commit to the merge base commit %s\n", mergeBaseCommit)
-			goodCommit = mergeBaseCommit
+			fmt.Printf("The old Bazel commit is not an ancestor of the good Bazel commit, overriding the old Bazel commit to the merge base commit %s\n", mergeBaseCommit)
```
Comment (meteorcloudy) [inline core/core.go:808]: good -> new?

**Review by meteorcloudy — APPROVED** [review]
Awesome, thanks for implementing this with such a simple syntax!

**Comment by fmeum** [comment]
@meteorcloudy Updated the PR. I started out with `!`, but then noticed that it is a special shell character that would have made it hard to use.

**Pull request merged**